components/drawer/index.tsx

```diff
@@ -0,0 +1,26 @@
+import { useState } from 'react'
+import { Drawer } from '@mui/material'
+import { StyledDrawer } from './StyledDrawer'
+import { useDispatch, useSelector } from 'react-redux'
+import { closeDrawer } from '@/lib/redux/slices/drawer'
+
+const FormDrawer = () => {
+    const dispatch = useDispatch()
+    const { drawerDisplay, showDrawer } = useSelector(
+        (state: any) => state.drawer
+    )
+
+    return (
+        <StyledDrawer>
+            <Drawer
+                anchor="right"
+                open={showDrawer}
+                PaperProps={{ style: { width: '50%' } }}>
+                <span onClick={() => dispatch(closeDrawer())}>Close</span>
```

Comment:
Lets turn this span into a div that takes up the full drawer. width, give it 1rem of height. Inside the div, lets have an icon button positioned at the right end with an X icon to close the drawer.